backend/src/main/java/gov/cdc/usds/simplereport/db/model/auxiliary/HttpRequestDetails.java

```diff
@@ -0,0 +1,96 @@
+package gov.cdc.usds.simplereport.db.model.auxiliary;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * The information that we capture about the HTTP request for an audit event. Must be stable (no
+ * type changes or field deletions) once released, since we will be using it to serialize records.
+ * Mostly we should expect the {{@link #getForwardedAddresses()} information to be what anybody ever
+ * consults, but the other stuff is easy to save here and hard to reconstruct otherwise.
+ */
+public class HttpRequestDetails {
+
+  /** Headers we extract from the request. */
+  private static final class Headers {
+    private static final String FORWARDED_PROTOCOL = "x-forwarded-proto";
+    private static final String FORWARDED_CLIENT = "x-forwarded-for";
+    private static final String FORWARDED_HOST = "x-original-host";
+  }
+
+  // all fields should be set in the constructor, so all fields are final.
+  private final String serverName;
+  private final String remoteAddress;
+  private final List<String> forwardedAddresses;
+  private final String forwardedProtocol;
+  private final String originalHostName;
+  private final String requestUri;
+
+  public HttpRequestDetails(HttpServletRequest request) {
+    serverName = request.getServerName(); // mostly irrelevant
+    originalHostName = request.getHeader(Headers.FORWARDED_HOST); // screw you, Azure
+    remoteAddress = request.getRemoteAddr();
+    ArrayList<String> headerNames = new ArrayList<>();
+    request.getHeaderNames().asIterator().forEachRemaining(headerNames::add);
+    String forwardedFor = request.getHeader(Headers.FORWARDED_CLIENT);
+    forwardedAddresses =
+        forwardedFor == null ? List.of() : Arrays.asList(forwardedFor.split(",\\s*"));
+    forwardedProtocol = request.getHeader(Headers.FORWARDED_PROTOCOL);
+    requestUri = request.getRequestURI();
+  }
+
+  @JsonCreator
+  public HttpRequestDetails(
+      @JsonProperty("serverName") String serverName,
+      @JsonProperty("remoteAddress") String remoteAddress,
+      @JsonProperty("forwardedAddresses") List<String> forwardedAddresses,
+      @JsonProperty("forwardedProtocol") String forwardedProtocol,
+      @JsonProperty("originalHostName") String originalHostName,
+      @JsonProperty("requestUri") String requestUri) {
+    super();
```

Comment:
Nitpicky: this class extends `Object`, so no need to call `super()`

Reply:
Dammit, I usually remember to delete those. 😆